commands/scp.go

```diff
@@ -148,6 +150,11 @@ func getInfoForScpArg(hostAndPath string, hostInfoLoader HostInfoLoader) (HostIn
 	}
 
 	args := []string{}
+	port, err := hostInfo.GetSSHPort()
+	if err == nil && port > 0 {
+		args = append(args, "-P", fmt.Sprintf("%v", port))
+	}
```

Comment:
I saw the blog, and tried without success. I don't think `-P` works for the multi remote host with different ports. Here is what I tried, with the debug turned on.

If I put the both `-P` at beginning, it seems only the last one works.

```bash
root@d2:~# scp -v -P 231 -P 232 root@138.68.61.69:./README.md root@138.68.50.80:./
Executing: /usr/bin/ssh '-x' '-oClearAllForwardings=yes' '-n' '-v' '-p' '231' '-p' '232' '-l' 'root' '--' '138.68.61.69' 'scp -v' './README.md' 'root@138.68.50.80:./'
OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g-fips  1 Mar 2016
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: /etc/ssh/ssh_config line 59: Applying options for *
debug1: Connecting to 138.68.61.69 [138.68.61.69] port 232.
debug1: connect to address 138.68.61.69 port 232: Connection refused
ssh: connect to host 138.68.61.69 port 232: Connection refused
```

If I put `-P` in front of each server, only the first server is ok to connect.

```bash
root@d2:~# scp -v -P 231 root@138.68.61.69:./README.md -P 232 root@138.68.50.80:./
Executing: /usr/bin/ssh '-x' '-oClearAllForwardings=yes' '-n' '-v' '-p' '231' '-l' 'root' '--' '138.68.61.69' 'scp -v -d' './README.md' 'root@138.68.50.80:./'
OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g-fips  1 Mar 2016
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: /etc/ssh/ssh_config line 59: Applying options for *
debug1: Connecting to 138.68.61.69 [138.68.61.69] port 231.
debug1: Connection established.
debug1: permanently_set_uid: 0/0
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_rsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_rsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_dsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_dsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_ecdsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_ecdsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_ed25519 type -1
debug1: key_load_public: No such file or directory
debug1: identity file /root/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.1
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.2p2 Ubuntu-4ubuntu2.1
debug1: match: OpenSSH_7.2p2 Ubuntu-4ubuntu2.1 pat OpenSSH* compat 0x04000000
debug1: Authenticating to 138.68.61.69:231 as 'root'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256@libssh.org
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:Y+0bgv/+1MhjHkZ7QX9TNMSfZMd1NMUsAvDKN0IB24Q
debug1: Host '[138.68.61.69]:231' is known and matches the ECDSA host key.
debug1: Found key in /root/.ssh/known_hosts:1
debug1: rekey after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: rekey after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS received
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<rsa-sha2-256,rsa-sha2-512>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: Trying private key: /root/.ssh/id_rsa
debug1: Authentication succeeded (publickey).
Authenticated to 138.68.61.69 ([138.68.61.69]:231).
debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: network
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug1: Sending environment.
debug1: Sending env LANG = en_US.UTF-8
debug1: Sending command: scp -v -d ./README.md root@138.68.50.80:./
Executing: program /usr/bin/ssh host 138.68.50.80, user root, command scp -v -d -t ./
OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g-fips  1 Mar 2016
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: /etc/ssh/ssh_config line 59: Applying options for *
debug1: Connecting to 138.68.50.80 [138.68.50.80] port 22.
debug1: connect to address 138.68.50.80 port 22: Connection refused
ssh: connect to host 138.68.50.80 port 22: Connection refused
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
lost connection
debug1: channel 0: free: client-session, nchannels 1
Transferred: sent 2400, received 3232 bytes, in 0.2 seconds
Bytes per second: sent 10566.2, received 14229.1
debug1: Exit status 1
Executing: program /usr/bin/ssh host 138.68.50.80, user root, command scp -v -d -t ./
OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g-fips  1 Mar 2016
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: /etc/ssh/ssh_config line 59: Applying options for *
debug1: Connecting to 138.68.50.80 [138.68.50.80] port 231.
debug1: connect to address 138.68.50.80 port 231: Connection refused
ssh: connect to host 138.68.50.80 port 231: Connection refused
lost connection
root@d2:~#
```